### PR TITLE
coap-gateway: owner cache integration into service

### PR DESCRIPTION
**Pull request by Danielius1922**

### CODE REVIEW

File: coap-gateway/config.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ apis:
     address: "0.0.0.0:5684"
     externalAddress: ""
     maxMessageSize: 262144
+    cacheExpiration: 10m
```
Comment (ondrejtomcik) [inline coap-gateway/config.yaml]: It's not descriptive enough. ownerCachceExpiration? Should it be really limited in the case of CoAP Gateway which should keep the cache open as long as the device is connected? There shouldn't be imho any time-based expiration.

Comment (jkralik) [reply coap-gateway/config.yaml]: It's nice when you connect `1000 devices` at the same time so it will use data from the updated cache.

Comment (ondrejtomcik) [reply coap-gateway/config.yaml]: Decrease to 1m

Comment (Danielius1922) [reply coap-gateway/config.yaml]: Updated

File: resource-aggregate/cqrs/eventbus/nats/subscriber/subscriber.go [hunk 2]
```diff
@@ -134,45 +135,57 @@ func WithGoPool(goroutinePoolGo eventbus.GoroutinePoolGoFunc) GoroutinePoolGoOpt
...
+		return nil, fmt.Errorf("cannot create cert manager: %w", err)
+	}
+	config.Options = append(config.Options, nats.Secure(certManager.GetTLSConfig()), nats.MaxReconnects(-1))
+	conn, err := nats.Connect(config.URL, config.Options...)
+	if err != nil {
+		certManager.Close()
+		return nil, fmt.Errorf("cannot create nats connection: %w", err)
+	}
+
+	s, err := NewWithNATS(conn, config.PendingLimits, logger, opts...)
```
Comment (jkralik) [inline resource-aggregate/cqrs/eventbus/nats/subscriber/subscriber.go]: who close conn function? `NewWithNATS` I expected that the caller close conn. And with `New` I expected that will be close by subscribers. 
Can we rename `NewWithNATS` to just `New` and drop the old `New`?

File: resource-aggregate/cqrs/eventbus/nats/publisher/publisher.go [hunk 3]
```diff
@@ -53,39 +54,49 @@ func WithMarshaler(dataMarshaler MarshalerFunc) MarshalerOpt {
...
 	}
 	p.AddCloseFunc(certManager.Close)
 	return p, nil
 }
 
-// NewPublisher creates a publisher.
-func newPublisher(url string, jetstream bool, eventMarshaler MarshalerFunc, options ...nats.Option) (*Publisher, error) {
-	conn, err := nats.Connect(url, options...)
+// Create publisher with existing NATS connection
+func NewWithNATS(conn *nats.Conn, jetstream bool, opts ...Option) (*Publisher, error) {
```
Comment (jkralik) [inline resource-aggregate/cqrs/eventbus/nats/publisher/publisher.go]: can we rename `NewWithNATS` to `New` and drop the old `New`?

File: resource-aggregate/cqrs/eventbus/nats/subscriber/subscriber.go (hunk 2 above)
Comment (Danielius1922) [reply resource-aggregate/cqrs/eventbus/nats/subscriber/subscriber.go]: The current `subscriber.New` is used 14 times and `publisher.New` is used 33 times (mostly in tests though), so when I get rid of the original `New` then some convenience method equivalent would be useful.

Comment (Danielius1922) [reply resource-aggregate/cqrs/eventbus/nats/subscriber/subscriber.go]: Solved by creating helper function for tests.

**Review by ondrejtomcik — APPROVED** [review]

**Review by jkralik — APPROVED** [review]

**Review by jkralik — APPROVED** [review]

**Pull request merged**